stat/stat.go

```diff
@@ -1001,6 +1004,23 @@ func Quantile(p float64, c CumulantKind, x, weights []float64) float64 {
 			}
 		}
 		panic("impossible")
+	case AvgDiscont:
+		var cumsum float64
+		fidx := p * sumWeights
+		for i := range x {
+			if weights == nil {
+				cumsum++
+			} else {
+				cumsum += weights[i]
+			}
+			if cumsum >= fidx {
+				if len(x)%2 != 0 {
```

Comment:
This is not right. It's not about the length of the slice being even or odd, it's about the percentile not being able to be met exactly. For instance, in []float64{1, 2}, you'd expect to get a cumulant of 1/3rd exactly. The smoothing needs to be about p_low and p_high (when you add the previous and next sample).